### ISSUE TITLE
artifact names are tricky

### ISSUE THREAD

**Issue by flozano** [qos-ch/logback-access#21]
If you depend on logback-access 'tomcat' module, for example, and you have a `libs/` directory with all your maven (or gradle) dependencies, you will have a 'jar' sitting there named 'tomcat-2.0.3.jar'. Thanks to having version number there, it's not going to collide with any other `tomcat-*.jar` (usually) - but it could, and in any case it can get confusing.

This is a common problem when namespacing the artifact names (which end up as jars in a dir) inside a groupId - not a new issue.

The problem is that fixing requires relocating the maven coordinates (adding `logback-access-` prefix to the artifact names).

**Comment by ceki** [comment]
@flozano  Indeed, artifact names are tricky. I agree naming logback-access:tomcat can be problematic and needs to be addressed.

**Comment by ceki** [comment]
Fixed in commit 4d87a3fd2131c1c1

**Comment by runeflobakk** [comment]
Just for reference to others:
This change of the naming scheme (for the better, I agree) _may_ result in incompatible versions of logback-access and the core Logback artifacts if you fail to upgrade Logback Access artifacts together with Logback core/classic. I upgraded Logback to 1.5.15, but did not get any notification of newer Logback Access, because I already used the latest artifacts of the previously named "common" and "jetty12", and had to track down the cause of a `ClassNotFoundException: ch.qos.logback.core.boolex.JaninoEventEvaluatorBase`.

As an extra precaution, I added the following exclusion to my `<bannedDependencies>` in my maven-enforcer-plugin configuration:
```xml
<!-- Not allow any of the early 2.x releases of Logback Access with previous naming scheme-->
<exclude>ch.qos.logback.access:*:[2,2.0.4)</exclude>
```
Just to avoid risking to end up with multiple versions of what is really the same dependency (e.g. "common" and "logback-access-common"), and unpredictable use of classes from the versions of these artifacts.

I was wondering (as I have no experience with this mechanism myself) is this an applicable case for "[relocation POMs](https://maven.apache.org/guides/mini/guide-relocation.html)" to be deployed for the previously named artifacts?
In my case, I run `mvn versions:display-dependency-updates`, would a relocation POM deployed as `ch.qos.logback.access:common:2.0.4` pointing to the new artifact have given me an appropriate warning that the artifact is now named logback-access-common?
